pkg/bundle/util.go

```diff
@@ -18,25 +18,27 @@ package bundle
 
 import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
 
 	trustapi "github.com/cert-manager/trust-manager/pkg/apis/trust/v1alpha1"
+	trustapiac "github.com/cert-manager/trust-manager/pkg/applyconfigurations/trust/v1alpha1"
 )
 
 // bundleHasCondition returns true if the bundle has an exact matching condition.
 // The given condition will have the ObservedGeneration set to the bundle Generation.
 // The LastTransitionTime is ignored.
 func bundleHasCondition(
```

Comment:
I feel like using the ApplyConfiguration object in this function really adds a lot of unnecessary logic.
I would like to keep using trustapi.BundleCondition here.
In general, I think there is no reason to have `trustapiac.BundleConditionApplyConfiguration`, since we will always atomically edit such a struct (and not its individual fields).

Reply:
> I feel like using the ApplyConfiguration object in this function really adds a lot of unnecessary logic.

Which logic is added? The logic is the same - we just have to be more explicit about zero-values, mainly because of the tests that don't set mandatory fields. I think this function should not exist, and instead patch unconditionally or have a more "technical" check if a patch is needed or not. IMO the established pattern of `needsUpdate` is fragile in general and adds unnecessary complexity - both code and tests.

> I would like to keep using trustapi.BundleCondition here.

Where would you then put the conversion logic? I feel this comment is more about your general objection to apply configurations commented below....

> In general, I think there is no reason to have trustapiac.BundleConditionApplyConfiguration, since we will always atomically edit such a struct (and not its individual fields).

I notice you are not a big fan of apply configurations, and that's a fair point of view. 😸 If we don't believe that's the way forward, we should probably remove the generation of them. I agree that AC's are not ideal, but I think it gives us some more freedom when modeling the resources Go API. Without them, we have to make sure we **always** use a pointer type - if not we risk breaking the SSA semantics.